src/main/java/org/humancellatlas/ingest/bundle/BundleManifestRepositoryImpl.java

```diff
@@ -0,0 +1,44 @@
+package org.humancellatlas.ingest.bundle;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.util.List;
+
+public class BundleManifestRepositoryImpl implements BundleManifestRepositoryCustom {
+
+    private final MongoTemplate mongoTemplate;
+
+    @Autowired
+    public BundleManifestRepositoryImpl(final MongoTemplate mongoTemplate){
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    @Override
+    public Page<BundleManifest> findBundles(String projectUuid, String primarySubmissionUuid, Boolean isPrimary, Pageable pageable) {
+        Query query = new Query();
+
+        query.addCriteria(Criteria.where("fileProjectMap." + projectUuid).exists(true));
+
+        if (isPrimary != null && isPrimary){
```

Comment:
If isPrimary == null should that be taken as automatically looking for the primary submission?

Reply:
Lol! what have I been thinking here?! Thanks Justin! This is a good suggestion! :) I'll apply this to the one in dev